dotcom-rendering/src/components/ArticleMeta.apps.tsx

```diff
@@ -0,0 +1,332 @@
+import { shouldShowAvatar, shouldShowContributor } from './ArticleMeta';
+import { Avatar } from './Avatar';
+import { Branding } from './Branding.importable';
+import { CommentCount } from './CommentCount.importable';
+import { Contributor } from './Contributor';
+import { Dateline } from './Dateline';
+import { FollowWrapper } from './FollowWrapper.importable';
+import { Island } from './Island';
+import { SendAMessage } from './SendAMessage.importable';
+
+type Props = {
+	format: ArticleFormat;
+	pageId: string;
+	webTitle: string;
+	byline?: string;
+	tags: TagType[];
+	primaryDateline: string;
+	secondaryDateline: string;
+	branding?: BrandingType;
+	discussionApiUrl: string;
+	shortUrlId: string;
+	isCommentable: boolean;
+	ajaxUrl: string;
+	messageUs?: MessageUs;
+};
+
+const metaGridContainer = css`
+	display: grid;
+	grid-template-rows: auto auto auto auto auto;
+	grid-template-columns: 10px auto 1fr auto 10px;
+	grid-template-areas:
+		' . . . . .'
+		'. avatar byline comment-count .'
+		'. avatar byline comment-count .'
+		' . . . . .'
+		'. dateline dateline dateline .';
+
+	${from.mobileLandscape} {
+		grid-template-columns: 20px auto 1fr auto 20px;
+	}
+
+	${from.phablet} {
+		grid-template-columns: 0px auto 1fr auto 0px;
+	}
+`;
+
+const metaContainerMargins = css`
+	${until.phablet} {
+		margin-left: -20px;
+		margin-right: -20px;
+	}
+	${until.mobileLandscape} {
+		margin-left: -10px;
+		margin-right: -10px;
+	}
+`;
+
+const MetaGridAvatar = ({ children }: { children: React.ReactNode }) => (
+	<div
+		css={css`
+			grid-area: avatar;
+			width: 60px;
+			height: 60px;
+			margin-top: ${space[2]}px;
+			margin-right: ${space[3]}px;
+			margin-bottom: ${space[3]}px;
+			margin-left: 0px;
+			${from.leftCol} {
+				width: 140px;
+				height: 140px;
+			}
+		`}
+	>
+		{children}
+	</div>
+);
+
+const MetaGridByline = ({
+	children,
+	isComment,
+}: {
+	children: React.ReactNode;
+	isComment: boolean;
+}) => (
+	// address > div > span > svg:first-of-type - spacing for the Follow Tag buttons in follow wrapper,
+	// which is contextual of the meta by layout type
+	<div
+		css={css`
+			grid-area: byline;
+			width: auto;
+			margin-top: ${space[2]}px;
+			margin-bottom: ${space[3]}px;
+
+			address > div {
+				padding-bottom: 0px;
+			}
+
+			address > div > span > svg:first-of-type {
+				display: block;
+				margin-top: ${space[2]}px;
+				${!isComment && `margin-bottom: ${space[2]}px;`}
+			}
+		`}
+	>
+		{children}
+	</div>
+);
+
+const MetaGridCommentCount = ({
+	children,
+	isPicture,
+	isComment,
+	isImmersive,
+	isAnalysis,
+	isLiveBlog,
+}: {
+	children: React.ReactNode;
+	isPicture: boolean;
+	isComment: boolean;
+	isImmersive: boolean;
+	isAnalysis: boolean;
+	isLiveBlog: boolean;
+}) => (
+	<div
+		data-print-content="hide"
+		css={css`
+			grid-area: comment-count;
+			border-left: 1px solid
+				${isLiveBlog
+					? 'rgba(255, 255, 255, 0.4)'
+					: themePalette('--article-meta-lines')};
```

Comment:
[Best practice in Emotion is to use the `style` prop for dynamic styles](https://emotion.sh/docs/best-practices#use-the-style-prop-for-dynamic-styles).

Can be addressed in a follow-up